dev/com.ibm.ws.jaxrs.2.1.common/src/com/ibm/ws/jaxrs21/sse/LibertySseEventSinkImpl.java

```diff
@@ -89,39 +90,48 @@ public boolean isClosed() {
     /* (non-Javadoc)
      * @see javax.ws.rs.sse.SseEventSink#send(javax.ws.rs.sse.OutboundSseEvent)
      */
+    @FFDCIgnore({WebApplicationException.class, IOException.class})
     @Override
     public CompletionStage<?> send(OutboundSseEvent event) {
         final CompletableFuture<?> future = new CompletableFuture<>();
 
         if (!closed) {
             if (writer != null) {
-                try (ByteArrayOutputStream os = new ByteArrayOutputStream()) {
+                ByteArrayOutputStream os = null;
+                try {
+                    os = new ByteArrayOutputStream();
                     writer.writeTo(event, event.getClass(), null, new Annotation [] {}, event.getMediaType(), null, os);
 
                     String eventContents = os.toString();
                     if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
                         Tr.debug(tc, "send - sending " + eventContents);
                     }
-                    
+
                     if (!response.isCommitted()) {
                         response.setHeader("Content-Type", MediaType.SERVER_SENT_EVENTS);
                         response.flushBuffer();
                     }
-                    
+
                     //TODO: this seems like a bug, but most SSE clients seem to expect a named event
                     //      so for now, we will provide one if one is not provided by the user
                     if (event.getName() == null) {
                         response.getOutputStream().print("    UnnamedEvent\n");
                     }
                     response.getOutputStream().println(eventContents);
                     response.getOutputStream().flush();
-                    
+
                     return CompletableFuture.completedFuture(eventContents);
-                } catch (WebApplicationException | IOException ex) {
-                    //TODO: convert to warning?
-                    if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                        Tr.debug(tc, "send - failed sending event " + event);
-                        future.completeExceptionally(ex);
+                } catch (WebApplicationException ex) {
+                    handleException(ex, future, event);
+                } catch (IOException ex) {
+                    handleException(ex, future, event);
+                } finally {
+                    if (os != null) {
```

Comment:
Shouldn't this be  "if (( os != null) || (os.size() == 0)) since you set o = new ByteArrayInputStream right away so I'm not sure when it would be "null".

Reply:
No, we always want to close the ByteArrayOutputStream.  The case where it could be null is if the constructor itself threw an exception.

Reply:
That makes sense... agreed.